test/integration/multiple.done.js

```diff
@@ -4,25 +4,95 @@ var args   = [];
 
 describe('multiple calls to done()', function() {
   var res;
-
   this.timeout(1000);
 
-  before(function(done) {
-    run('multiple.done.js', args, function(err, result) {
-      res = result;
-      done(err);
+  describe('from a spec', function() {
+    before(function(done) {
+      run('multiple.done.js', args, function(err, result) {
+        res = result;
+        done(err);
+      });
+    });
+
+    it('results in failures', function() {
+      assert.equal(res.stats.pending, 0);
+      assert.equal(res.stats.passes, 1);
+      assert.equal(res.stats.failures, 1);
+      assert.equal(res.code, 1);
+    });
+
+    it('throws a descriptive error', function() {
+      assert.equal(res.failures[0].err.message,
+        'done() called multiple times');
+    });
+  });
+
+  describe('with multiple specs', function() {
+    before(function(done) {
+      run('multiple.done.specs.js', args, function(err, result) {
+        res = result;
+        done(err);
+      });
+    });
+
+    it('results in a failure', function() {
+      assert.equal(res.stats.pending, 0);
+      assert.equal(res.stats.passes, 2);
+      assert.equal(res.stats.failures, 1);
+      assert.equal(res.code, 1);
+    });
+
+    it('correctly attributes the error', function() {
+      assert.equal(res.failures[0].fullTitle, 'suite test1');
+      assert.equal(res.failures[0].err.message,
+        'done() called multiple times');
     });
   });
 
-  it('results in failures', function() {
-    assert.equal(res.stats.pending, 0);
-    assert.equal(res.stats.passes, 1);
-    assert.equal(res.stats.failures, 1);
-    assert.equal(res.code, 1);
+  describe('from a before hook', function() {
+    before(function(done) {
+      run('multiple.done.before.js', args, function(err, result) {
+        res = result;
+        done(err);
+      });
+    });
+
+    it('results in a failure', function() {
+      assert.equal(res.stats.pending, 0);
+      assert.equal(res.stats.passes, 1);
```

Comment:
> I mean, we obviously could ignore it, but what do you think about doing so?

I don't think it's ideal. I think that leads to the same confusing output we see in scenarios like:

``` javascript
$ cat test.js
beforeEach(function(done) {
  done(new Error('error'));
});

it('test1', function(done) {
  setTimeout(done, 20);
});

it('test1', function(done) {
  setTimeout(done, 20);
});

$ mocha test.js


  1) "before each" hook for "test1"

  0 passing (11ms)
  1 failing

  1)  "before each" hook for "test1":
     Error: error
      at Context.<anonymous> (test.js:2:8)
```